src/psyclone/tests/psyir/transformations/arraynotation2arrayrange_trans_test.py

```diff
@@ -0,0 +1,199 @@
+
+    '''
+    trans = ArrayNotation2ArrayRangeTrans()
+    psyir = fortran_reader.psyir_from_source(code)
+    for reference in psyir.walk(Reference):
+        try:
+            trans.apply(reference)
+        except TransformationError:
+            pass
+    return fortran_writer(psyir)
+
+
+def test_get_array_bound(fortran_reader):
+    ''' Test that the get_array_bound utility works as expected '''
+    # known bounds
+    psyir = fortran_reader.psyir_from_source(CODE)
+    node = psyir.walk(Reference)[0]
+    symbol = node.symbol
+    lower_bound, upper_bound, step = \
+        ArrayNotation2ArrayRangeTrans._get_array_bound(symbol, 0)
+    from psyclone.psyir.nodes import Literal, BinaryOperation
+    assert isinstance(lower_bound, Literal)
+    assert lower_bound.value == "1"
+    assert isinstance(upper_bound, Literal)
+    assert upper_bound.value == "10"
+    assert isinstance(step, Literal)
+    assert step.value == "1"
+    # unknown bounds
+    psyir = fortran_reader.psyir_from_source(
+        CODE.replace("dimension(10)", "dimension(:)"))
+    node = psyir.walk(Reference)[0]
+    symbol = node.symbol
+    lower_bound, upper_bound, step = \
+        ArrayNotation2ArrayRangeTrans._get_array_bound(symbol, 0)
+    assert isinstance(lower_bound, BinaryOperation)
+    assert lower_bound.operator == BinaryOperation.Operator.LBOUND
+    assert isinstance(upper_bound, BinaryOperation)
+    assert upper_bound.operator == BinaryOperation.Operator.UBOUND
+    assert isinstance(step, Literal)
+    assert step.value == "1"
+
+
+def test_transform():
+    '''Check that it is possible to create an instance of
+    ArrayNotation2ArrayRangeTrans and that it is a Transformation.
+
+    '''
+    assert ArrayNotation2ArrayRangeTrans()
+    assert isinstance(ArrayNotation2ArrayRangeTrans(), Transformation)
+
+
+def test_notation(fortran_reader, fortran_writer):
+    '''Test that array notation gets replaced with an array range when the
+    size of the array range is known.
+
+    '''
+    result = apply_trans(fortran_reader, fortran_writer, CODE)
+    assert "a(1:10) = b\n" in result
+
+
+def test_dimension(fortran_reader, fortran_writer):
+    '''Test that array notation gets replaced with an array range when the
+    size of the array dimension is unknown.
+
+    '''
+    code = CODE.replace("dimension(10)", "dimension(:)")
+    result = apply_trans(fortran_reader, fortran_writer, code)
+    assert "a(:) = b\n" in result
+
+
+def test_variable(fortran_reader, fortran_writer):
+    '''Test that a array notation gets replaced with an array range when
+    the size of the array dimension is a variable.
+
+    '''
+    code = CODE.replace("  real, dimension(10) :: a\n",
+                        "  integer :: n\n  real, dimension(n) :: a\n")
+    result = apply_trans(fortran_reader, fortran_writer, code)
+    assert "a(1:n) = b\n" in result
+
+
+def test_range(fortran_reader, fortran_writer):
+    '''Test that an array slice remains unchanged.'''
```

Comment:
As mentioned below, I think this really belongs in the validate test.

Reply:
Added new validate test based on this test. Kept this test as is. Let me know if you think it should be removed.